src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3332,7 +3380,8 @@ protected Object capitalizeSingleByte(RubyString string, int caseMappingOptions,
                 finalBytes[0] ^= 0x20;
             }
 
-            string.setTString(fromByteArrayNode.execute(finalBytes, string.encoding.tencoding)); // codeRangeNode.execute(rope), characterLengthNode.execute(rope)
+            string.setTString(makeTString(finalBytes, encoding));
```

Comment:
This doesn't look correct, given the `byteArray` can have an offset and `byte[].clone` is used in `invertAsciiCaseNode`

Reply:
I need to take another pass over `invertAsciiCaseNode`. I'll keep this in mind. If it's a problem, so is the call on `finalBytes[0] ^= 0x20`. The thinking was that `copyByteArray` makes an exact size copy.

Reply:
Both your comment and mine were addressed by using `TruffleString.CopyToByteArrayNode`, which will ensure the array is the exact length we need and indices are relative to a 0 offset.